### PR TITLE
Issue #1691 - Improves installation script

### PR DESCRIPTION
**Pull request by ajainuary**
As described in #1691 the PR includes some changes which allow for checking exit code before presenting the success message.

### CODE REVIEW

File: initial_setup.sh [hunk 1]
```diff
@@ -1,32 +1,44 @@
-#!/bin/bash
```
Comment (qgolsteyn) [inline initial_setup.sh]: We should keep the shebang in the script
https://stackoverflow.com/questions/13872048/bash-script-what-does-bin-bash-mean

File: initial_setup.sh [hunk 2]
```diff
@@ -1,32 +1,44 @@
...
-ln -s `pwd`/node_modules frog
-ln -s frog/babel.config.js .
-node linkFiles.js
+if python -c 'import sys; print(sys.version_info[0] == 2)' > /dev/null == 'True';
+then
+    YARN_VERSION='1.15.2'
+    if which yarn && [[ `yarn --version` == $YARN_VERSION ]]; then
+        echo 'Using pre-installed global Yarn'; YARN=yarn
+    else
+        if [ -f "./node_modules/.bin/yarn" ] && [[ `./node_modules/.bin/yarn --version` == $YARN_VERSION ]]; then
```
Comment (qgolsteyn) [inline initial_setup.sh]: What happens if the script is not run from the correct directory?

File: initial_setup.sh [hunk 3]
```diff
@@ -1,32 +1,44 @@
...
+    fi
+    echo "Yarn: $YARN"
 
-echo
-echo -e "\xE2\x9C\xA8  Finished Initial Setup."
-echo -e "\xF0\x9F\xA4\x93  Run ${LBLUE}npm start server ${NC}to begin hacking!"
+    if ! "$YARN" install; then
+    	echo "Sorry, there seems to be something wrong, please correct the errors and try again"
+    	echo "There was some error installing the required packages, please rectify the errors and try again"
+	    exit 1
```
Comment (qgolsteyn) [inline initial_setup.sh]: We should be a bit more descriptive with our error message. Maybe something along the lines of "Unable to run yarn install successfully. You might not be connected to the internet, or there might be a problem with your FROG folder"

File: initial_setup.sh [hunk 4]
```diff
@@ -1,32 +1,44 @@
...
-    echo 'Using pre-installed global Yarn'; YARN=yarn
-else
-    if [ -f "$FROG/node_modules/.bin/yarn" ] && [[ `"$FROG/node_modules/.bin/yarn" --version` == $YARN_VERSION ]]; then
-        echo 'Using pre-installed local Yarn'; YARN="$FROG/node_modules/.bin/yarn"
-    else
-        echo 'Installing Yarn'; npm install yarn@$YARN_VERSION --no-package-lock && YARN="$FROG/node_modules/.bin/yarn"
-    fi
+if ! PYTHON_VERSION=$(python --version);
+then
+    echo "Cannot find Python, please install python 2 and try again"
```
Comment (qgolsteyn) [inline initial_setup.sh]:
``` suggestion
    echo "Cannot find Python, please install Python 2 and try again"
```

File: initial_setup.sh [hunk 5]
```diff
@@ -1,32 +1,44 @@
...
+    fi
+    rm -rf frog/node_modules
+    ln -s `pwd`/node_modules frog
+    ln -s frog/babel.config.js .
+    node linkFiles.js
 
-exit 0
+    echo
+    echo -e "\xE2\x9C\xA8  Finished Initial Setup."
+    echo -e "\xF0\x9F\xA4\x93  Run ${LBLUE}npm start server ${NC}to begin hacking!"
```
Comment (qgolsteyn) [inline initial_setup.sh]: nit: We could move the emojis to constants to help with readability

File: initial_setup.sh [hunk 6]
```diff
@@ -1,32 +1,44 @@
...
+    exit 0
+else
```
Comment (qgolsteyn) [inline initial_setup.sh]: This else statement could be changed to an if statement checking if the script failed and exiting early.

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
Left a couple of comments.

Generally:
- Would be good to add comments to describe the various commands in this script
- Maybe we should clean up the script to remove nested if statements. Try to keep error checking contained and exit early.

File: initial_setup.sh (hunk 1 above)
Comment (ajainuary) [reply initial_setup.sh]: Sorry, I didn't mean to remove it. It was an inadvertent error.

File: initial_setup.sh (hunk 3 above)
Comment (ajainuary) [reply initial_setup.sh]: That would be displayed by `yarn` itself. This just informs the user that his action wasn't successful.

File: initial_setup.sh (hunk 2 above)
Comment (ajainuary) [reply initial_setup.sh]: > go to the root repository

It is mentioned in the instructions for `README.md`
We could have error handling for that but seems a bit far-fetched idea to me.

File: initial_setup.sh (hunk 5 above)
Comment (ajainuary) [reply initial_setup.sh]: We could introduce a flag to remove emoji.

File: initial_setup.sh (hunk 6 above)
Comment (ajainuary) [reply initial_setup.sh]: Made the respective changes

File: initial_setup.sh (hunk 2 above)
Comment (qgolsteyn) [reply initial_setup.sh]: Good point, maybe add a quick comment in the file itself to keep this assumption close to the source code

File: initial_setup.sh (hunk 5 above)
Comment (qgolsteyn) [reply initial_setup.sh]: Unclear if there is benefit in doing so, this file will normally only be run once

**Review by qgolsteyn — APPROVED** [review]
Looks good! We could add some comments indicating the various steps we go through in this script for readability. Something like:

```
## INSTALLING PYTHON ##

<code goes here>
```

Very nitpicky so feel free to ignore if you don't have time

**Pull request merged**